client/my-sites/checkout/cart/header-cart.jsx

```diff
@@ -31,15 +30,18 @@ class HeaderCart extends React.Component {
 	};
 
 	render() {
-		if ( isEmpty( getAllCartItems( this.props.cart ) ) ) {
-			return null;
+		const isCartEmpty = getAllCartItems( this.props.cart ).length === 0;
+		const noop = () => {};
```

Comment:
Just a quick comment that it might be better to use [lodash's `noop`](https://lodash.com/docs/4.17.15#noop). Or at least define this one as a method on this component. Otherwise, react could think it's always a new prop and re-render the cart.

Reply:
That's an interesting point! I had removed it because I think it's a good idea to remove lodash anywhere we can, since it heavily uses its own functions and can add a lot to bundle sizes, but as this does re-create the function on each render of `HeaderCart`, it may unnecessarily re-render `PopoverCart`, but only if `PopoverCart` is memoized (passed through `connect()` or `React.memo()` or implements `shouldComponentUpdate`), which is does not. If a component is not memoized, it will always be rendered when its parent is re-rendered, even if its props do not change.

If a component's output does not change, React will not update the DOM, so the main reason that I know of to avoid extra renders is expensive side effects, but I don't see any such side effects in `PopoverCart`, and this change shouldn't cause more renders in any case.

That said, it really seems to me that `closeSectionNavMobilePanel` should be optional on `PopoverCart` (the PropTypes on that component list it as optional, but it's actually required). I may push a simple change as part of this PR so that it doesn't need a noop at all.

Reply:
TIL! Thanks for the in-depth explanation 🙇

Reply:
For completeness: there's one other case I thought of that can prevent rendering if the props don't change: if the component extends `React.PureComponent`.